app/controllers/addresses_controller.rb

```diff
@@ -0,0 +1,40 @@
+class AddressesController < ApplicationController
+  def edit
+    @address = Address.find(params[:id])
+    respond_to do |format|
+      format.js { render "edit_address_form" }
+    end
+  end
+
+  def update
+    address = Address.find(params[:id])
+    if address.update_attributes(address_params)
+      redirect_to address_summary_orders_path(address.id)
```

Comment:
Exception Handling! What if update attribute fails what happens. You might want to redirect to the edit page?